GANDLF/entrypoints/hf_hub_integration.py

```diff
@@ -0,0 +1,131 @@
+from GANDLF.cli.huggingface_hub_handler import push_to_model_hub, download_from_hub
+
+
+@click.command()
+@click.option(
+    "--upload/--download",
+    "-u/-d",
+    required=True,
+    help="Upload or download to/from a Huggingface Repo",
+)
+@click.option(
+    "--repo-id",
+    "-rid",
+    required=True,
+    help="Downloading/Uploading: A user or an organization name and a repo name separated by a /",
+)
+@click.option(
+    "--token",
+    "-tk",
+    help="Downloading/Uploading: A token to be used for the download/upload",
+)
+@click.option(
+    "--revision",
+    "-rv",
+    help="Downloading/Uploading: git revision id which can be a branch name, a tag, or a commit hash",
+)
+@click.option(
+    "--cache-dir",
+    "-cdir",
+    help="Downloading: path to the folder where cached files are stored",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.option(
+    "--local-dir",
+    "-ldir",
+    help="Downloading: if provided, the downloaded file will be placed under this directory",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.option(
+    "--force-download",
+    "-fd",
+    is_flag=True,
+    help="Downloading: Whether the file should be downloaded even if it already exists in the local cache",
+)
+@click.option(
+    "--folder-path",
+    "-fp",
+    help="Uploading: Path to the folder to upload on the local file system",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.option(
+    "--path-in-repo",
+    "-pir",
+    help="Uploading: Relative path of the directory in the repo. Will default to the root folder of the repository",
+)
+@click.option(
+    "--commit-message",
+    "-cr",
+    help='Uploading: The summary / title / first line of the generated commit. Defaults to: f"Upload {path_in_repo} with huggingface_hub"',
+)
+@click.option(
+    "--commit-description",
+    "-cd",
+    help="Uploading: The description of the generated commit",
+)
+@click.option(
+    "--repo-type",
+    "-rt",
+    help='Uploading: Set to "dataset" or "space" if uploading to a dataset or space, "model" if uploading to a model. Default is model',
+)
+@click.option(
+    "--allow-patterns",
+    "-ap",
+    help="Uploading: If provided, only files matching at least one pattern are uploaded.",
+)
+@click.option(
+    "--ignore-patterns",
+    "-ip",
+    help="Uploading: If provided, files matching any of the patterns are not uploaded.",
+)
+@click.option(
+    "--delete-patterns",
+    "-dp",
+    help="Uploading: If provided, remote files matching any of the patterns will be deleted from the repo while committing new files. This is useful if you don't know which files have already been uploaded.",
+)
+@click.option(
+    "--hf-template",
+    "-hft",
+    help="Adding the template path for the model card it is Required during Uploaing a model",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
```

Comment:
So we have to provide a default path for the hugging face template so that  when ever  the user want to upload a ,model by default that template will be uploaded ?

Reply:
That's what I suggest to do yes. There is already a hugging_face.md template in this PR so I was suggesting to reuse it.

Reply:
Ok thanks for you suggestion  @Wauplin we  will do  the same